src/backend_webgl2compute.ts

```diff
@@ -399,6 +399,78 @@ export class WebGL2ComputeBackend extends KernelBackend {
     return this.compileAndRun(program, [x], output);
   }
 
+  fromPixels(
+      pixels: backend_util.PixelData|ImageData|HTMLImageElement|
+      HTMLCanvasElement|HTMLVideoElement,
+      numChannels: number): Tensor3D {
+    if (pixels == null) {
+      throw new Error(
+          'pixels passed to tf.browser.fromPixels() can not be null');
+    }
+
+    const outShape = [pixels.height, pixels.width, numChannels];
+    let imageData = (pixels as ImageData | backend_util.PixelData).data;
+
+    if (ENV.getBool('IS_BROWSER')) {
+      console.log(1);
+      if (!(pixels instanceof HTMLVideoElement) &&
+          !(pixels instanceof HTMLImageElement) &&
+          !(pixels instanceof HTMLCanvasElement) &&
+          !(pixels instanceof ImageData) &&
+          !((pixels as backend_util.PixelData).data instanceof Uint8Array)) {
+        throw new Error(
+            'pixels passed to tf.browser.fromPixels() must be either an ' +
+            `HTMLVideoElement, HTMLImageElement, HTMLCanvasElement, ImageData` +
+            ` or {data: Uint32Array, width: number, height: number}, ` +
+            `but was ${(pixels as {}).constructor.name}`);
+      }
+      if (pixels instanceof HTMLVideoElement) {
+        if (this.fromPixels2DContext == null) {
+          this.fromPixels2DContext =
+              document.createElement('canvas').getContext('2d');
+          this.fromPixels2DContext.canvas.width = pixels.width;
+          this.fromPixels2DContext.canvas.height = pixels.height;
+        }
+        this.fromPixels2DContext.drawImage(
+            pixels, 0, 0, pixels.width, pixels.height);
+        pixels = this.fromPixels2DContext.canvas;
+      }
+
+      // TODO: Remove this once we figure out how to upload textures directly to
+      // WebGPU.
+      const imageDataLivesOnGPU = pixels instanceof HTMLVideoElement ||
+          pixels instanceof HTMLImageElement ||
+          pixels instanceof HTMLCanvasElement;
+      if (imageDataLivesOnGPU) {
+        imageData = this.fromPixels2DContext
+                        .getImageData(0, 0, pixels.width, pixels.height)
+                        .data;
+      }
+    }
+
+    // TODO: Encoding should happen on GPU once we no longer have to download
+    // image data to the CPU.
+    let pixelArray = imageData;
+    if (numChannels != null && numChannels !== 4) {
+      pixelArray = new Uint8Array(pixels.width * pixels.height * numChannels);
+
+      for (let i = 0; i < imageData.length; i++) {
+        if (i % 4 < numChannels) {
+          const pixelIndex = Math.floor(i / 4);
+          pixelArray[pixelIndex * numChannels + i % 4] = imageData[i];
+        }
+      }
+    }
+
+    const output = this.makeOutputArray(outShape, 'int32');
+    this.write(output.dataId, Int32Array.from(pixelArray));
+    return output as Tensor3D;
+  }
+
+  cast<T extends Tensor>(x: T, dtype: DataType): T {
+    return backend_util.castTensor(x, dtype, this);
+  }
```

Comment:
```
  cast<T extends Tensor>(x: T, dtype: DataType): T {
    return backend_util.castTensor(x, dtype, this);
  }
```
`cast` is an standalone operator. It shouldn't be included in `fromPixels`.

Reply:
Done